### PR TITLE
vcv-rack: add desktop item

### PR DESCRIPTION
**Pull request by ddelabru**
###### Motivation for this change

VCV Rack is distributed for Linux platforms as a source or binary tarball with no desktop entry. Because it is a desktop application, I have created a desktop item for it here.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111042` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>vcv-rack</li>
</ul>
</details>

**Review by Moredread — APPROVED** [review]
I can't easily see the resulting Desktop Items as I'm using i3 as my wm, but the source of the changes looks good.

**Comment by doronbehar** [comment]
@ddelabru this PR looks good, though there are merge conflicts now. Could you address them please? BTW, tip: To avoid such merge conflicts, using 1 item per list in the arguments of `buildInputs` and in the arguments at the beginning of the expression helps a lot.

File: pkgs/applications/audio/vcv-rack/default.nix [hunk 1]
```diff
@@ -40,6 +41,18 @@ with lib; stdenv.mkDerivation rec {
...
+    type = "Application";
+    name = pname;
+    desktopName = "VCV Rack";
+    genericName = "Eurorack simulator";
+    comment = "Create music by patching together virtual synthesizer modules";
+    exec = "Rack";
+    icon = "Rack";
+    categories = "AudioVideo;AudioVideoEditing;Audio;";
+    extraEntries = "Keywords=music;";
+  }) ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/vcv-rack/default.nix:54]:
```suggestion
  desktopItems = [
    (makeDesktopItem {
      type = "Application";
      name = pname;
      desktopName = "VCV Rack";
      genericName = "Eurorack simulator";
      comment = "Create music by patching together virtual synthesizer modules";
      exec = "Rack";
      icon = "Rack";
      categories = "AudioVideo;AudioVideoEditing;Audio;";
      extraEntries = "Keywords=music;";
    })
  ];
```

File: pkgs/applications/audio/vcv-rack/default.nix [hunk 2]
```diff
@@ -87,6 +101,20 @@ with lib; stdenv.mkDerivation rec {
...
     # Override the default global resource file directory
     wrapProgram $out/bin/Rack --add-flags "-s $out/share/vcv-rack"
+
+    # Extract pngs from the Apple icon image and create
+    # the missing ones from the 1024x1024 image.
+    icns2png --extract icon.icns
+    for size in 16 24 32 48 64 128 256 512 1024; do
+      mkdir -pv $out/share/icons/hicolor/"$size"x"$size"/apps
+      if [ ! -e icon_"$size"x"$size"x32.png ]
+      then
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/vcv-rack/default.nix:111]:
```suggestion
      if [ ! -e icon_"$size"x"$size"x32.png ]; then
```

File: pkgs/applications/audio/vcv-rack/default.nix [hunk 3]
```diff
@@ -87,6 +101,20 @@ with lib; stdenv.mkDerivation rec {
...
+        convert -resize "$size"x"$size" icon_1024x1024x32.png icon_"$size"x"$size"x32.png
+      fi
+      install -Dm644 icon_"$size"x"$size"x32.png $out/share/icons/hicolor/"$size"x"$size"/apps/Rack.png
+    done;
+
+    copyDesktopItems
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/vcv-rack/default.nix:117]:
```suggestion
    runHook postInstall
```

Also please add the preHook.

**Comment by doronbehar** [comment]
Closing due to no response from OP.

**Pull request closed without merging**